scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua

```diff
@@ -1,194 +1,528 @@
+        else
+            -- use non-dverger 2-hour at 50% hp of current phase
+            if phaseSpecialID > 0 then
+                local halfHP = mobPhaseHP[phase] / 2
+                if mobHP < halfHP then
+                    mob:useMobAbility(phaseSpecialID)
+                    mob:setLocalVar('usedSpecial', 1)
+                end
             end
-        end
-
-        -- Increment astral
-        mob:setLocalVar('astralFlow', astral + 1)
 
-    -- Or, at least make sure pets weren't drug off
-    else
-        --[[ Unused
-        for i = 1, 8 do
-            local pet = nil
-            if phase == 21 then
-                pet = pets[astral % 2][i]
-            else
-                pet = pets[phase % 2][i]
+            -- make sure pets weren't dragged off...
+            for i = 0, 1 do
+                for j = 1, 8 do
+                    local pet = GetMobByID(petIDs[i][j])
+                    if pet:isAlive() and not pet:isEngaged() then
+                        pet:updateEnmity(target)
+                    end
+                end
             end
         end
-        ]]--
     end
 
     -- Check for time limit, too
     if os.time() > depopTime and mob:actionQueueEmpty() then
-        for i = 0, 1 do
-            for j = 1, 8 do
-                if pets[i][j]:isSpawned() then
-                    DespawnMob(petIDs[i][j])
-                end
-            end
-        end
-
         DespawnMob(ID.mob.PANDEMONIUM_WARDEN)
+    --  printf('Timer expired at %i. Despawning Pandemonium Warden.', depopTime)
     end
 end
 
 entity.onMobDeath = function(mob, player, optParams)
-    player:addTitle(xi.title.PANDEMONIUM_QUELLER)
+    if optParams.isKiller or optParams.noKiller then
+        mob:showText(mob, ID.text.PW_END_OF_NOTHING)
+    end
 
-    -- Despawn pets
-    for i = 0, 1 do
-        for j = 1, 8 do
-            if GetMobByID(petIDs[i][j]):isSpawned() then
-                DespawnMob(petIDs[i][j])
-            end
-        end
+    if player ~= nil then
+        player:addTitle(xi.title.PANDEMONIUM_QUELLER)
     end
+
+    pwFuncs.despawnPets()
 end
 
 entity.onMobDespawn = function(mob)
-    -- Despawn pets
-    for i = 0, 1 do
-        for j = 1, 8 do
-            if GetMobByID(petIDs[i][j]):isSpawned() then
-                DespawnMob(petIDs[i][j])
+    pwFuncs.despawnPets()
+end
+
+entity.onCriticalHit = function(mob)
+    local phase = mob:getLocalVar('phase')
+    if phase == 18 then -- Hydra phase
+        local critNum = mob:getLocalVar('crits')
+
+        if critNum + 1 > mob:getLocalVar('CritToTheFace') then  -- Lose a head
+            if mob:getAnimationSub() == 0 then
+                mob:setAnimationSub(1)
+                mob:setLocalVar('headTimer', os.time() + math.random(60, 190))
+            elseif mob:getAnimationSub() == 1 then
+                mob:setAnimationSub(2)
+                mob:setLocalVar('headTimer', os.time() + math.random(60, 190))
+            else
+                -- Meh
```

Comment:
why the empty logical branch?

Reply:
Great question...

Reply:
I'm assuming i had used that slot for debugging prints and just replaced it with "Meh" when i went live on Wings